src/luac-cli.js

```diff
@@ -0,0 +1,438 @@
+        process.stderr.write(Buffer.from((`${progname}: unrecognized option '${message}'\n`)));
+    else
+        process.stderr.write(Buffer.from((`${progname}: ${message}\n`)));
+
+    process.stderr.write(Buffer.from((
+        `usage: ${progname} [options] [filename]\n`
+        + `Available options are:\n`
+        + `  -l       list (use -l -l for full listing)\n`
+        + `  -o name  output to file 'name' (default is \"%{OUTPUT}\")\n`
+        + `  -p       parse only\n`
+        + `  -s       strip debug information\n`
+        + `  -v       show version information\n`
+        + `  --       stop handling options\n`
+        + `  -        stop handling options and process stdin\n`
+    )));
+
+    process.exit(1);
+};
+
+const doargs = function() {
+    let version = 0;
+    let i;
+    for (i = 2; i < process.argv.length; i++) {
+        let arg = process.argv[i];
+
+        if (arg[0] !== "-")  /* end of options; keep it */
+            break;
+
+        else if (arg === "--") {  /* end of options; skip it */
+            ++i;
+            if (version) ++version;
+            break;
+        } else if (arg === "-")   /* end of options; use stdin */
+            break;
+        else if (arg === "-l")  /* list */
+            ++listing;
+        else if (arg === "-o") {  /* output file */
+            output = process.argv[++i];
+            if (!output || (output[0] == "-" && output.length > 1))
+                usage("'-o' needs argument");
+            if (arg === "-") output = null;
+        } else if (arg === "-p")  /* parse only */
+            dumping = false;
+        else if (arg === "s")  /* strip debug information */
+            stripping = true;
+        else if (arg === "-v")  /* show version */
+            ++version;
+        else  /* unknown option */
+            usage(arg);
+    }
+
+    if (i === process.argv.length && (listing || !dumping)) {
+        dumping = false;
+        process.argv[--i] = OUTPUT;
+    }
+
+    if (version) {
+        print(lua.LUA_COPYRIGHT);
+        if (version === process.argv.length - 2) process.exit(0);
+    }
+    return i;
+};
+
+const FUNCTION = "(function()end)();";
+
+const reader = function(L, ud) {
+    if (ud--)
+        return lua.to_luastring(FUNCTION);
+    else
+        return null;
+};
+
+const combine = function(L, n) {
+    if (n === 1)
+        return lua.lua_topointer(L, -1).p;
+    else {
+        let i = n;
+        if (lua.lua_load(L, reader, i, lua.to_luastring(`=(${PROGNAME})`), null) !== lua.LUA_OK)
+            fatal(lua.lua_tojsstring(L, -1));
+
+        let f = lua.lua_topointer(L, -1).p;
+        for (i = 0; i < n; i++) {
+            f.p[i] = lua.lua_topointer(L, i - n - 1).p;
+            if (f.p[i].upvalues.length > 0)
+                f.p[i].upvalues[0].instack = false;
+        }
+        f.lineinfo.length = 0;
+        return f;
+    }
+};
+
+const writer = function(L, p, size, u) {
+    return fs.writeSync(u, new Buffer(p), 0, size) > 0 ? 0 : 1;
```

Comment:
Yes and `new Buffer` is even [deprecated](https://nodejs.org/api/buffer.html#buffer_new_buffer_array)